src/UNL/UCBCN/Manager/AddDatetime.php

```diff
@@ -0,0 +1,165 @@
+<?php
+namespace UNL\UCBCN\Manager;
+
+use UNL\UCBCN\Calendar as CalendarModel;
+use UNL\UCBCN\Event;
+use UNL\UCBCN\Locations;
+use UNL\UCBCN\Event\Occurrence;
+
+class AddDatetime implements PostHandlerInterface 
+{
+	public $options = array();
+    public $calendar;
+    public $event;
+    public $event_datetime;
+
+    public function __construct($options = array()) 
+    {
+        $this->options = $options + $this->options;
+        $this->calendar = CalendarModel::getByShortName($this->options['calendar_shortname']);
+        if ($this->calendar === FALSE) {
+            throw new \Exception("That calendar could not be found.", 404);
+        }
+
+        $this->event = Event::getByID($this->options['event_id']);
+        if ($this->event === FALSE) {
+            throw new \Exception("That event could not be found.", 404);
+        }
+
+        if (array_key_exists('event_datetime_id', $this->options)) {
+            # we are editing an existing datetime
+            $this->event_datetime = Occurrence::getByID($this->options['event_datetime_id']);
+
+            if ($this->event_datetime === FALSE) {
+                throw new \Exception("That datetime could not be found", 404);
+            }
+        } else {
+            # we are adding a new datetime
+            $this->event_datetime = new Occurrence;
+        }
+    }
+
+    public function getUserLocations()
+    {
+        $user = Auth::getCurrentUser();
+        return new Locations(array('user_id' => $user->uid));
+    }
+    
+    public function getStandardLocations($display_order)
+    {
+        return new Locations(array(
+            'standard'      => true,
+            'display_order' => $display_order,
+        ));
+    }
+
+    private function calculateDate($date, $hour, $minute, $am_or_pm)
+    {
+        $date = strtotime($date);
+        # add hours correctly
+        # TODO: handle when this is not entered
```

Comment:
Just a reminder that there is still a todo here.  Did you want to do this before this is merged?